### PR TITLE
Define individuals for spatial resolution

### LINKED ISSUE #15: Define additional terms for spatial resolution in the GeoDCAT-AP namespace?
> :information_source: Following from background discussion in https://github.com/SEMICeu/GeoDCAT-AP/issues/3

DCAT-AP 2 includes two properties from DCAT 2 to specify spatial and temporal resolution:
- [`dcat:spatialResolutionInMeters`](https://www.w3.org/TR/vocab-dcat-2/#Property:dataset_spatial_resolution)
- [`dcat:temporalResolution`](https://www.w3.org/TR/vocab-dcat-2/#Property:dataset_temporal_resolution)

They will therefore be eventually inherited by the future version of GeoDCAT-AP - ideally, taking into account backward compatibility.

It is worth noting, however, that `dcat:spatialResolutionInMeters` cannot express spatial resolution as equivalent scale, so an option would be defining a specific property in the GeoDCAT-AP namespace.

Based on the DCAT 2 approach, this property could be named

`geodcatap:spatialResolutionAsEquivalentScale`

or, shortly:

`geodcatap:spatialResolutionAsScale`

and take as value an `xsd:decimal` expressing the fraction. E.g.:

````turtle
a:Dataset a dcat:Dataset ;
  geodcatap:spatialResolutionAsScale "0.001"^^xsd:decimal ;
.
````
would state that the spatial resolution of the dataset in equivalent scale is 1:1,000. 

This solution can also be extended to other types of spatial resolution:
- `geodcatap:spatialResolutionAsAngularDistance` or, shortly,  `geodcatap:spatialResolutionInDegrees`
- `geodcatap:spatialResolutionAsVerticalDistanceInMeters` (yes, that's terribly long)

The corresponding tentative definitions:

````turtle
geodcatap:spatialResolutionAsScale a rdf:Property ;
  rdfs:label "spatial resolution as equivalent scale"@en ;
  rdfs:comment """Spatial resolution expressed as equivalent scale, 
    measured by using a representative fraction (e.g., 1:1,000, 1:1,000,000)."""@en ;
  rdfs:range xsd:decimal ;
 .

geodcatap:spatialResolutionInDegrees a rdf:Property ;
  rdfs:label "spatial resolution as angular distance"@en ;
  rdfs:comment """Spatial resolution expressed as angular distance, 
    measured in decimal degrees."""@en ;
  rdfs:range xsd:decimal ;
 .

geodcatap:spatialResolutionAsVerticalDistanceInMeters a rdf:Property ;
  rdfs:label "spatial resolution as vertical distance"@en ;
  rdfs:comment """Spatial resolution expressed as vertical distance, 
    measured in metres."""@en ;
  rdfs:range xsd:decimal ;
 .
````

### PR DESCRIPTION
**Pull request by andrea-perego**
See https://github.com/SEMICeu/GeoDCAT-AP/issues/15

Summary of changes:
- Add definition of the relevant individual in the RDF
- Add relevant DQV classes and properties in the RDF
- Update specification accordingly

Preview: https://raw.githack.com/SEMICeu/GeoDCAT-AP/spatial-resolution-dqv/drafts/latest/index.html

### CODE REVIEW

File: drafts/latest/examples/metric.ttl [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+geodcatap:spatialResolutionAsDistance a dqv:Metric ;
+  skos:prefLabel "Spatial resolution as distance"@en ;
+  dqv:expectedDataType xsd:decimal ;
+  dqv:inDimension dqv:precision .
+
+geodcatap:spatialResolutionAsScale a dqv:Metric ;
+  skos:prefLabel "Spatial resolution as equivalent scale"@en ;
+  dqv:expectedDataType xsd:decimal ;
+  dqv:inDimension <dqv:precision> .
```
Comment (riccardoAlbertoni) [inline drafts/latest/examples/metric.ttl]:
```suggestion
  dqv:inDimension dqv:precision .
```

File: drafts/latest/tables/deprecated-properties-for-data-service.html [hunk 2]
```diff
@@ -91,7 +97,9 @@
 
 <aside class="note">
 <p>Until GeoDCAT-AP version 1.0.1 [[GeoDCAT-AP-20160802]], property <code>rdfs:comment</code> was used with class <a href="#service"><code>dctype:Service</code></a> (replaced by class <code>dcat:DataService</code> since this version of GeoDCAT-AP) to specify the spatial resolution of a Data Service, since a more specific property was not available in reference vocabularies.</p>
-<p>[[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>. Therefore <code>dcat:spatialResolutionInMeters</code> replaces <code>rdfs:comment</code> starting from this version of GeoDCAT-AP, but only when spatial resolution is expressed as distance in metres.</p>
+<p>[[VOCAB-DCAT-2]] and [[?SDW-BP]] recommend the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, via property <code>dcat:hasQualityMeasurement</code>. Moreover, [[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>
```
Comment (riccardoAlbertoni) [inline drafts/latest/tables/deprecated-properties-for-data-service.html]:
```suggestion
<p>[[VOCAB-DCAT-2]] and [[?SDW-BP]] recommend the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, via property <code>dqv:hasQualityMeasurement</code>. Moreover, [[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>
```

File: drafts/latest/tables/deprecated-properties-for-dataset.html [hunk 3]
```diff
@@ -35,5 +36,7 @@
 
 <aside class="note">
 <p>Until GeoDCAT-AP version 1.0.1 [[GeoDCAT-AP-20160802]], property <code>rdfs:comment</code> was used with class <code>dcat:Dataset</code> to specify the spatial resolution of a Dataset, since a more specific property was not available in reference vocabularies.</p>
-<p>[[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>, which has been adopted by [[DCAT-AP-20191120]]. Therefore, <code>dcat:spatialResolutionInMeters</code> replaces <code>rdfs:comment</code> starting from this version of GeoDCAT-AP, but only when spatial resolution is expressed as distance in metres.</p>
+<p>[[VOCAB-DCAT-2]] and [[?SDW-BP]] recommend the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, via property <code>dcat:hasQualityMeasurement</code>. Moreover, [[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>
```
Comment (riccardoAlbertoni) [inline drafts/latest/tables/deprecated-properties-for-dataset.html]:
```suggestion
<p>[[VOCAB-DCAT-2]] and [[?SDW-BP]] recommend the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, via property <code>dqv:hasQualityMeasurement</code>. Moreover, [[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>
```

Comment (riccardoAlbertoni) [inline drafts/latest/tables/deprecated-properties-for-dataset.html]: [[VOCAB-DCAT-2]]  refers to DQV for expressing quality statements, but AFAIR, it does not explicitly refer to dqv:hasQualityMeasurement for spatial resolution.
Besides correcting  "dcat:hasQualityMeasurement" with "dqv:hasQualityMeasurement", I would consider rephrasing the note a little ...

```suggestion
<p>[[?SDW-BP]] recommends the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, via property <code>dqv:hasQualityMeasurement</code>.  [[VOCAB-DCAT-2]] reuses  [[VOCAB-DQV]]  for expressing quality statements and it  has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>
```

File: drafts/latest/tables/deprecated-properties-for-distribution.html [hunk 4]
```diff
@@ -35,5 +36,7 @@
 
 <aside class="note">
 <p>Until GeoDCAT-AP version 1.0.1 [[GeoDCAT-AP-20160802]], property <code>rdfs:comment</code> was used with class <code>dcat:Distribution</code> to specify the spatial resolution of a Distribution, since a more specific property was not available in reference vocabularies.</p>
-<p>[[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>, which has been adopted by [[DCAT-AP-20191120]]. Therefore <code>dcat:spatialResolutionInMeters</code> replaces <code>rdfs:comment</code> starting from this version of GeoDCAT-AP, but only when spatial resolution is expressed as distance in metres.</p>
+<p>[[VOCAB-DCAT-2]] and [[?SDW-BP]] recommend the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, via property <code>dcat:hasQualityMeasurement</code>. Moreover, [[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>
```
Comment (riccardoAlbertoni) [inline drafts/latest/tables/deprecated-properties-for-distribution.html]:
```suggestion
<p>[[VOCAB-DCAT-2]] and [[?SDW-BP]] recommend the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, via property <code>dqv:hasQualityMeasurement</code>. Moreover, [[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>
```

Comment (riccardoAlbertoni) [inline drafts/latest/tables/deprecated-properties-for-distribution.html]: [[VOCAB-DCAT-2]]  refers to DQV for expressing quality statements, but AFAIR, it does not explicitly refer to dqv:hasQualityMeasurement for spatial resolution.
Besides correcting  "dcat:hasQualityMeasurement" with "dqv:hasQualityMeasurement", I would consider rephrasing the note a little ...

```suggestion
<p>[[?SDW-BP]] recommends the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, via property <code>dqv:hasQualityMeasurement</code>.  [[VOCAB-DCAT-2]] reuses  [[VOCAB-DQV]]  for expressing quality statements and it  has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>
```

File: drafts/latest/tables/deprecated-properties-for-data-service.html (hunk 2 above)
Comment (riccardoAlbertoni) [inline drafts/latest/tables/deprecated-properties-for-data-service.html]: [[VOCAB-DCAT-2]]  refers to DQV for expressing quality statements, but AFAIR, it does not explicitly refer to dqv:hasQualityMeasurement for spatial resolution.
Besides correcting  "dcat:hasQualityMeasurement" with "dqv:hasQualityMeasurement", I would consider rephrasing the note a little ...

```suggestion
<p>[[?SDW-BP]] recommends the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, via property <code>dqv:hasQualityMeasurement</code>.  [[VOCAB-DCAT-2]] reuses  [[VOCAB-DQV]]  for expressing quality statements and it  has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>
```

File: drafts/latest/tables/recommended-properties-for-metric.html [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+</td>
+<td>
+<p><code>dqv:inDimension</code></p>
+</td>
+<td>
+<p><code>dqv:Dimension</code></p>
+</td>
+<td>
+<p>This property represents the dimensions a quality metric, certificate and annotation allow a measurement of.</p>
+<p>In GeoDCAT-AP, this property is used in the definition of the instances of <code>dqv:Metric</code> corresponding to the different type of spatial resolution. It MUST always take as value <code>dqv:precision</code>.</p>
```
Comment (riccardoAlbertoni) [inline drafts/latest/tables/recommended-properties-for-metric.html]: What if in the future or in some profile of GeoDCAT-AP, DQV is used also for other quality dimensions? I would consider contextualizing more the 'MUST', for example, rephasing as in the following. 
```suggestion
<p>In GeoDCAT-AP, this property is used in the definition of the instances of <code>dqv:Metric</code> corresponding to the different type of spatial resolution. When referring to the different type of spatial resolution, it MUST always take as value <code>dqv:precision</code>.</p>
```

**Review by riccardoAlbertoni — COMMENTED** [review]
Some minor suggestions which might be considered.

File: drafts/latest/examples/metric.ttl (hunk 1 above)
Comment (andrea-perego) [reply drafts/latest/examples/metric.ttl]: Thanks for spotting this typo, @riccardoAlbertoni . The bug was in the .rdf file, which is used as master to generate also the .ttl. I fixed both.

File: drafts/latest/tables/deprecated-properties-for-data-service.html (hunk 2 above)
Comment (andrea-perego) [reply drafts/latest/tables/deprecated-properties-for-data-service.html]: Thanks again. I fixed the typo throughout the whole document.

File: drafts/latest/tables/deprecated-properties-for-dataset.html (hunk 3 above)
Comment (andrea-perego) [reply drafts/latest/tables/deprecated-properties-for-dataset.html]: Thanks again. I fixed the typo throughout the whole document.

Comment (andrea-perego) [reply drafts/latest/tables/deprecated-properties-for-dataset.html]: The referred statement is the one at the end of [§6.6.4 Property: spatial resolution](https://www.w3.org/TR/vocab-dcat-2/#Property:dataset_spatial_resolution) of DCAT2 - quoting:

> The range of this property is a decimal number representing a length in meters. This is intended to provide a summary indication of the spatial resolution of the data as a single number. More complex descriptions of various aspects of spatial precision, accuracy, resolution and other statistics can be provided using the Data Quality Vocabulary [VOCAB-DQV]. 

There is indeed no explicit reference to `dqv:hasQualityMeasurement`, but the [DQV section about prevision and accuracy](https://www.w3.org/TR/vocab-dqv/#ExpressDatasetAccuracyPrecision) is using that property in the examples showing how to specify spatial resolution. Said that, the current wording is indeed ambiguous, so maybe we can replace "via property `dqv:hasQualityMeasurement`" with "where property `dqv:hasQualityMeasurement` is used".

The resulting revision:

> <p>[[VOCAB-DCAT-2]] and [[?SDW-BP]] recommend the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, where property <code>dcat:hasQualityMeasurement</code> is used. Moreover, [[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>

 @riccardoAlbertoni , would this make sense to you?

File: drafts/latest/tables/deprecated-properties-for-distribution.html (hunk 4 above)
Comment (andrea-perego) [reply drafts/latest/tables/deprecated-properties-for-distribution.html]: Thanks again. I fixed the typo throughout the whole document.

Comment (andrea-perego) [reply drafts/latest/tables/deprecated-properties-for-distribution.html]: The referred statement is the one at the end of [§6.6.4 Property: spatial resolution](https://www.w3.org/TR/vocab-dcat-2/#Property:dataset_spatial_resolution) of DCAT2 - quoting:

> The range of this property is a decimal number representing a length in meters. This is intended to provide a summary indication of the spatial resolution of the data as a single number. More complex descriptions of various aspects of spatial precision, accuracy, resolution and other statistics can be provided using the Data Quality Vocabulary [VOCAB-DQV]. 

There is indeed no explicit reference to `dqv:hasQualityMeasurement`, but the [DQV section about prevision and accuracy](https://www.w3.org/TR/vocab-dqv/#ExpressDatasetAccuracyPrecision) is using that property in the examples showing how to specify spatial resolution. Said that, the current wording is indeed ambiguous, so maybe we can replace "via property `dqv:hasQualityMeasurement`" with "where property `dqv:hasQualityMeasurement` is used".

The resulting revision:

> <p>[[VOCAB-DCAT-2]] and [[?SDW-BP]] recommend the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, where property <code>dcat:hasQualityMeasurement</code> is used. Moreover, [[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>

 @riccardoAlbertoni , would this make sense to you?

File: drafts/latest/tables/deprecated-properties-for-data-service.html (hunk 2 above)
Comment (andrea-perego) [reply drafts/latest/tables/deprecated-properties-for-data-service.html]: The referred statement is the one at the end of [§6.6.4 Property: spatial resolution](https://www.w3.org/TR/vocab-dcat-2/#Property:dataset_spatial_resolution) of DCAT2 - quoting:

> The range of this property is a decimal number representing a length in meters. This is intended to provide a summary indication of the spatial resolution of the data as a single number. More complex descriptions of various aspects of spatial precision, accuracy, resolution and other statistics can be provided using the Data Quality Vocabulary [VOCAB-DQV]. 

There is indeed no explicit reference to `dqv:hasQualityMeasurement`, but the [DQV section about prevision and accuracy](https://www.w3.org/TR/vocab-dqv/#ExpressDatasetAccuracyPrecision) is using that property in the examples showing how to specify spatial resolution. Said that, the current wording is indeed ambiguous, so maybe we can replace "via property `dqv:hasQualityMeasurement`" with "where property `dqv:hasQualityMeasurement` is used".

The resulting revision:

> <p>[[VOCAB-DCAT-2]] and [[?SDW-BP]] recommend the use of [[VOCAB-DQV]] for the specification of different types of spatial resolution, where property <code>dcat:hasQualityMeasurement</code> is used. Moreover, [[VOCAB-DCAT-2]] has defined a specific property to specify spatial resolution as distance in metres, namely, <code>dcat:spatialResolutionInMeters</code>.</p>

 @riccardoAlbertoni , would this make sense to you?

File: drafts/latest/tables/recommended-properties-for-metric.html (hunk 5 above)
Comment (andrea-perego) [reply drafts/latest/tables/recommended-properties-for-metric.html]: Good point, @riccardoAlbertoni . That was actually the intended meaning, but the wording was bad.

File: drafts/latest/tables/deprecated-properties-for-dataset.html (hunk 3 above)
Comment (riccardoAlbertoni) [reply drafts/latest/tables/deprecated-properties-for-dataset.html]: Yes, it works for me. Thanks.

File: drafts/latest/tables/deprecated-properties-for-data-service.html (hunk 2 above)
Comment (riccardoAlbertoni) [reply drafts/latest/tables/deprecated-properties-for-data-service.html]: Yes, it works for me. Thanks.

**Pull request merged**